server/core/terraform/async_client.go

```diff
@@ -107,8 +109,14 @@ func (c *AsyncClient) RunCommandAsyncWithInput(ctx context.Context, prjCtx comma
 		}()
 		go func() {
 			s := bufio.NewScanner(stderr)
+			buf := []byte{}
+			s.Buffer(buf, BufioScannerBufferSize)
+
 			for s.Scan() {
 				message := s.Text()
+				if c.logLineShouldBeStripped(message) {
```

Comment:
this function is responsible for too much. I wouldn't be surprised if there was other post processing usecases we might want, let's design this with that in mind.

I would prefer something like delegate output handler that runs a set of post processors before forwarding to the underlying handler. 

ie. something like:

```go
type PostProcessingLogHandler struct {
   regexesToFilter []regex
   delegate            ProjectCommandOutputHandler
}

func (h *PostProcessingLogHandler) Send(ctx ProjectCommandContext, line string) {
  
  for _, r := range h.regexesToFIlter {
      if r.match(line) {
          return
      }
  }

  h.delegate(ctx, line)
}

```

Reply:
Ok, but I think we would have to modify your Send() fxn here to return some kind of indicator (probably a boolean) for the original Async Client caller to use. That way, it can still determine if we should send that output line to the `outCh` for plan/apply results output.

https://github.com/lyft/atlantis/pull/275/files#diff-decf7341abe2a036ddf4f1be97a5d07026d91cccf8769128a254b4cf4bf20f36R120